lib/storage.js

```diff
@@ -572,6 +573,38 @@ Storage.prototype.fetchActiveAddresses = function(walletId, cb) {
   });
 };
 
+Storage.prototype.storeFiatRate = function(providerName, rates, cb) {
+  var self = this;
+
+  var now = Date.now();
+  async.each(rates, function(rate, next) {
+    self.db.collection(collections.FIAT_RATES).insert({
+      provider: providerName,
+      ts: now,
+      code: rate.code,
+      value: rate.value,
+    }, {
+      w: 1
+    }, next);
+  }, cb);
+};
+
+Storage.prototype.fetchFiatRate = function(providerName, code, ts, cb) {
+  var self = this;
+  self.db.collection(collections.FIAT_RATES).find({
+    provider: providerName,
+    code: code,
+    ts: {
+      $lte: ts
```

Comment:
The response sent to the client contains a `fetchedOn` field which should let the client app decide on the validity of the rate. Anyways, I think you are right that we have to limit this server-side also.